README.md

```diff
@@ -149,24 +151,84 @@ loadConfig[Conf](conf)
 ```
 
 
-## Customizing naming conventions
+## Extend the library to support new types
+
+Not all types are supported automatically by `pureconfig`. For instance, classes
+that are not case classes are not supported out-of-the-box:
+
+```scala
+> class Foo(var bar: Int) { override def toString: String = s"Foo(${bar.toString})" }
+> loadConfig[Foo]
+could not find implicit value for parameter conv: pureconfig.ConfigConvert[Foo]
+```
+
+`pureconfig` can be extended to support those types. To do so, an instance for the
+`ConfigConvert` typeclass must be provided implicitly, like
+
+```scala
+> implicit val foocc = ConfigConvert.stringConvert[Foo](s => Try(new Foo(s.toInt)), foo => foo.bar.toString)
+> loadConfig[Foo]
+Foo(1)
+```
+
+
+## Override behaviour for types
+
+It is possible to override the behaviour of `pureconfig` for a certain type by
+implementing another instance of the `ConfigConvert` typeclass. For instance,
+the default behaviour of `pureconfig` for `String` is to return the string itself
+in the configuration:
+
+```scala
+> import import com.typesafe.config.ConfigValueFactory
+> ConfigConvert[String].from(ConfigValueFactory.fromAnyRef("FooBar"))
+util.Try[String] = Success(FooBar)
+```
+
+Now let's say that we want to override this behaviour such that `String`s are
+always read lower case. We can do:
+
+```scala
+> implicit val overridestrcc = ConfigConvert.fromString(s => Try(s.toLowerCase))
+> ConfigConvert[String].from(ConfigValueFactory.fromAnyRef("FooBar"))
+util.Try[String] = Success(foobar)
+```
+
+
+## Override behaviour for case classes
+
+`pureconfig` has to assume some default conventions and behaviours when
+deriving `ConfigConvert` instances for case classes:
+
+- How do keys in config objects map to field names of the case class?
+- Are unknown keys allowed in the config object?
+- Should default values in case class fields be applied when its respective
+  config key is missing?
+  
+By default, `pureconfig` assumes that config keys are written in kebab case
```

Comment:
Suggestion: I liked how you laid out the questions above as bullet points. I'd lay out the answers here in the say way, one per bullet point above.